### PR TITLE
clippy: Fix warnings in `components/script/dom` & `components/servo`

### PR DESCRIPTION
**Pull request by taniishkaaa**
Fixes warnings in `components/script/dom` & `components/servo` as a part of #31500 

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes are a part of #31500 
- [X] These changes do not require tests because they do not modify functionality

### CODE REVIEW

File: components/script/dom/htmlformelement.rs [hunk 1]
```diff
@@ -88,6 +88,7 @@ use crate::task_source::TaskSource;
 #[derive(Clone, Copy, JSTraceable, MallocSizeOf, PartialEq)]
 pub struct GenerationId(u32);
 
+#[allow(clippy::type_complexity)]
```
Comment (mukilan) [inline components/script/dom/htmlformelement.rs:91]: Does this need to be at the level of the whole `struct`? Can we not have allow it just for `past_names_map`?

Comment (taniishkaaa) [reply components/script/dom/htmlformelement.rs:91]: Of course, I can do that. I wasn't sure if it'd make for cleaner code.

Comment (mukilan) [reply components/script/dom/htmlformelement.rs]: It is better to have it closer to the source of the warning, so we don't inadvertently allow new members of this struct with complex types.

Comment (mukilan) [reply components/script/dom/htmlformelement.rs]: About the "cleaner code", I assume here you mean the visual noise from the `allow` directive. I guess we can just add blank lines before and after the field ([example](https://github.com/servo/servo/blob/main/components/script/dom/htmlformelement.rs#L105)) so it doesn't seem too "crammed".

Comment (taniishkaaa) [reply components/script/dom/htmlformelement.rs]: okay great! I'll add the blank lines :)

Comment (taniishkaaa) [reply components/script/dom/htmlformelement.rs]: but won't `./mach test-tidy` report errors after doing that?

Comment (mukilan) [reply components/script/dom/htmlformelement.rs]: I tried it and it didn't.

**Review by mukilan — APPROVED** [review]

**Review by mukilan — APPROVED** [review]

**Pull request merged**